packages/api/src/routes/mint.ts

```diff
@@ -76,14 +80,42 @@ const mint: FastifyPluginAsync = async (fastify, opts): Promise<void> => {
           .send(new Error(`Player has no id (key: ${key})`))
       }
 
-      const web3 = new Web3()
+      const web3 = new Web3(new Web3.providers.HttpProvider(WEB3_PROVIDER))
       // Check address is valid
       if (!web3.utils.isAddress(request.body.address)) {
         return reply
           .status(409)
           .send(new Error(`Mint address should be a valid Ethereum addresss`))
       }
 
+      // Fetch metadata from contract using Web3
+      const abi = WITTY_BUFFICORNS_ERC721_ABI
+
+      const contract = new web3.eth.Contract(abi, WITMON_ERC721_ADDRESS)
+      // TODO: ranch id goes from 0 to 5, ensure compatibility with solidity
+      const ranchId = RANCH_TO_INDEX[player.ranch]
+
+      let callResult
+      try {
+        callResult = await contract.methods.getRanchWeather(ranchId).call()
+      } catch (err) {
+        console.error(err)
+        return reply
+          .status(404)
+          .send(new Error(`Error calling getRanchWeather method from contract`))
+      }
+
+      /* Returns: (
+            uint256 _lastTimestamp,
+            string memory _lastDescription
+        )
+        */
+      const ranchWeather = callResult[1]
+      console.log('ranch weather', ranchWeather)
```

Comment:
the ranch weather description is just being logged, as it is not required neither as a minting parameter, nor as an input for generating the minting backend signature

Reply:
therefore, there's no actual need to call `getRanchWeather(uint256)`, neither to cache its value, within the implementation of the 'mint' endpoint